src/parser/hunter/marksmanship/modules/spells/AimedShot.tsx

```diff
@@ -37,14 +36,14 @@ class AimedShot extends Analyzer {
   constructor(options: Options) {
     super(options);
 
-    this.addEventListener(EventEmitter.catchAll, this.onEvent);
+    this.addEventListener(Events.any, this.onEvent);
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.AIMED_SHOT), this.onCast);
     this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.TRUESHOT, SPELLS.DEAD_EYE_BUFF]), (event: ApplyBuffEvent) => this.onAffectingBuffChange(event));
     this.addEventListener(Events.refreshbuff.by(SELECTED_PLAYER).spell([SPELLS.TRUESHOT, SPELLS.DEAD_EYE_BUFF]), (event: RefreshBuffEvent) => this.onAffectingBuffChange(event));
     this.addEventListener(Events.removebuff.to(SELECTED_PLAYER).spell([SPELLS.TRUESHOT, SPELLS.DEAD_EYE_BUFF]), (event: RemoveBuffEvent) => this.onAffectingBuffChange(event));
   }
 
-  onEvent(event: any) {
+  onEvent(event: Event<any>) {
```

Comment:
Didn't @Dambroda add AnyEvent for this?

Reply:
AnyEvent doesnt work here as the eventtype for the catchall filter is EventType.Event, which maps to its own type, so TS complains about the function having the wrong parameter. as far as I know we can't make EventType.Event map to AnyEvent within the MappedEventTypes type, as it would be self-referencing

Reply:
If you add `[EventType.Event]: Event<'event'>,` to the mapping will it work as intended?

Reply:
nah because then you're limited to an event type that has no content (so just timestamp and type, that's it) as the parameter for your function, so you can't do other checks for example to see  what fields exist (that's basically what putro and I discussed in discord the other day)

Reply:
okay I get what you mean now (as per discord); thought you meant something else. I'll adapt as such